allium/lib/network_health.py

```
@@ -252,21 +252,94 @@ def _integrate_aroi_validation(health_metrics, relay_set, total_relays_count):
+        else:
+            # Cache miss — compute fresh.
+
+            # Schema-version handshake: warn once on unknown schemas;
+            # expose the version downstream so the dashboard can render
+            # a footnote. progress_logger may be a ProgressLogger
+            # instance (with .log()) or None — wrap into a plain
+            # callable so check_schema_version /
+            # _log_aroi_build_summary can stay logger-agnostic.
+            _logger_obj = getattr(relay_set, 'progress_logger', None)
+            if _logger_obj is not None and hasattr(_logger_obj, 'log'):
+                log_callable = lambda m: _logger_obj.log(m, increment_step=False)
```

Ruff (0.15.12)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `log_callable` as a `def`

(E731)